File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.php
class Solution {

    /**
     * @param Integer[] $prices
     * @return Integer
     */
    function maxProfit($prices) {
        $minPrice = current($prices);
        $maxProfit = 0;
        
        foreach($prices as $price){
            $minPrice = min($minPrice, $price);
            $maxProfit = max($maxProfit, $price - $minPrice);
        }
        
        return $maxProfit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    minPrice := prices[0]
    maxProfit := 0
    
    for i := 1; i < len(prices); i++ {
        minPrice = min(minPrice, prices[i])
        maxProfit = max(maxProfit, prices[i] - minPrice)
    }
    
    return maxProfit
}

func min(i, j int) int {
    if i < j {
        return i
    }
    
    return j
}

func max(i, j int) int {
    if i > j {
        return i
    }
    
    return j
}